src/compiler/core_public.rs

```diff
@@ -0,0 +1,17 @@
+pub struct SortedArray<TItem> {
+    pub _vec: Vec<TItem>,
+}
+
+impl<TItem> SortedArray<TItem> {
+    pub fn new(vec: Vec<TItem>) -> Self {
+        Self { _vec: vec }
+    }
+
+    pub fn push(&mut self, item: TItem) {
+        self._vec.push(item);
+    }
```

Comment:
The functions on `SortedArray` seem like they should enforce that it's sorted. (ie. `new` should sort the incoming `Vec` and `push` should insert the new item at the correct index.)

Reply:
Yup they will, so far I haven't implemented any of the actual sorting logic